exercises/arrays/product/product.js

```diff
@@ -10,13 +10,23 @@
  */
 function product(array) {
```

Comment:
updating so that if array.length === 0, then return 0.

Reply:
Not quite!

If you concatenate `array1` to `[]` you get back `array1`. That means...

```js
product(array1) === product(concat([], array1)) === product([]) * product(array1)
```

which means you'd have to have `product([]) === 1`. And in fact, if you do that, everything "just works" without any `if` statements in sight:

```js
function product(array) {
  let result = 1;

  for (let item of array) {
    result = result * item;
  }

  return result;
}
```

Reply:
For similar reasons, you'd want `sum([])` to be `0` so that

```js
sum(concat(array1, array2)) === sum(array1) + sum(array2)
```

for any two arrays.

Then you get:

```js
function sum(array) {
  let result = 0;

  for (let item of array) {
    result = result + item;
  }

  return result;
}
```

Again, everything fits together "for free", so to speak.

Reply:
Fun exercise.

Consider a function like `min` or `max` that finds the smallest or largest number in an array of numbers. Using the same rationale, what would `min([])` be?  That is, what would you fill in the blank below with to make everything "just work"?

```js
function smaller(x, y) {
  if (x < y) {
    return x;
  } else {
    return y;
  }
}

function min(array) {
  let result = _____;

  for (let item of array) {
    result = smaller(result, item);
  }

  return result;
}
```

Likewise, what about `max([])`?